### ISSUE TITLE
[Bug] "Expected this condition to be compile time known." when checking `platform.onyx`

### ISSUE THREAD

**Issue by somebody1234** [onyx-lang/onyx-lsp#3]
```c
    #if (typeof MAIN_FUNCTION) == #type () -> void { // An error on this line means no 'main.main' was found in your program.
        MAIN_FUNCTION();
```

Also, the only highlighted part is the `#` - ideally, either the LSP should return the entire condition, or the diagnostics should include an end index?

**Comment by brendanfh** [comment]
Yes, I agree. Currently the LSP simply compiles your program, and parses the output of the `onyx check` command. There needs to be better handling of multi-token errors (like this case), but currently only the single `#` token is used to tell you where the error is. When the compiler's error reporting is improved, the LSP will automatically be made better.

I will leave this issue open until that has been addressed in the main repository.
